app/scripts/background.js

```diff
@@ -475,6 +477,45 @@ function setupController(initState, initLangCode) {
     extension.browserAction.setBadgeBackgroundColor({ color: '#037DD6' });
   }
 
+  notificationManager.on(
+    NOTIFICATION_MANAGER_EVENTS.POPUP_CLOSED,
+    rejectUnapprovedNotifications,
+  );
+
+  function rejectUnapprovedNotifications() {
+    // Metamask can open a popup for a login, we need to check if
+    // we're on a confirmation page
+    if (controller.isUnlocked()) {
+      Object.keys(
+        controller.txController.txStateManager.getUnapprovedTxList(),
+      ).forEach((txId) =>
+        controller.txController.txStateManager.setTxStatusRejected(txId),
+      );
+      controller.messageManager.messages
+        .filter((msg) => msg.status === 'unapproved')
+        .forEach((tx) => controller.messageManager.rejectMsg(tx.id));
+      controller.personalMessageManager.messages
+        .filter((msg) => msg.status === 'unapproved')
+        .forEach((tx) => controller.personalMessageManager.rejectMsg(tx.id));
+      controller.decryptMessageManager.messages
+        .filter((msg) => msg.status === 'unapproved')
+        .forEach((tx) => controller.decryptMessageManager.rejectMsg(tx.id));
+      controller.encryptionPublicKeyManager.messages
+        .filter((msg) => msg.status === 'unapproved')
+        .forEach((tx) => controller.decryptMessageManager.rejectMsg(tx.id));
+      controller.typedMessageManager.messages
+        .filter((msg) => msg.status === 'unapproved')
+        .forEach((tx) => controller.decryptMessageManager.rejectMsg(tx.id));
+      Object.keys(
+        controller.approvalController.state.pendingApprovals,
+      ).forEach((approvalId) =>
+        controller.approvalController.reject(approvalId, new Error()),
```

Comment:
This empty error isn't very useful in describing why the request was rejected. This error gets returned to sites awaiting on a response, so we should ensure it matches the provider error spec and that it's informative.

Should we be using the approval controller directly here? (cc @rekmarks ) Right now we're only using this for permission requests, so we could reject each permission request instead via `permissionController.rejectPermissionRequest`, which would set the appropriate error.

In general I don't imagine we'd want to use the approval controller directly, because it's too generic. The controllers making use of it have more context for how to handle rejections and confirmations.

Reply:
I'll change it to permissionController for now